mcs/class/System/System.Net/HttpWebRequest.cs

```diff
@@ -835,11 +832,43 @@ void AddRange (string rangeSpecifier, long from, long to)
 			webHeaders.ChangeInternal ("Range", r);
 		}
 
-		
-		public override IAsyncResult BeginGetRequestStream (AsyncCallback callback, object state) 
+		WebOperation SendRequest (bool redirecting, BufferOffsetSize writeBuffer, CancellationToken cancellationToken)
+		{
+			lock (locker) {
+				WebConnection.Debug ($"HWR SEND REQUEST: Req={ID} {requestSent} {redirecting}");
+
+				WebOperation operation;
+				if (!redirecting) {
+					if (requestSent) {
+						operation = currentOperation;
+						if (operation == null)
+							throw new InvalidOperationException ("Should never happen!");
+						return operation;
+					}
+				}
+
+				operation = new WebOperation (this, writeBuffer, false, cancellationToken);
+				if (Interlocked.CompareExchange (ref currentOperation, operation, null) != null)
+					throw new InvalidOperationException ("Invalid nested call.");
+
+				requestSent = true;
+				if (!redirecting)
+					redirects = 0;
+				servicePoint = GetServicePoint ();
+				servicePoint.SendRequest (operation, connectionGroup);
+				return operation;
+			}
+		}
+
+		Task<Stream> MyGetRequestStreamAsync ()
```

Comment:
It'd be better to inline this, it's used from a single place anyway

Reply:
Done.